Fix compiler warnings

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Motivation

When I compile the latest code, I found the following warning:

```
kvrocks/src/commands/cmd_cluster.cc:314:11: warning: private field 'slot_' is not used [-Wunused-private-field]
int64_t slot_ = -1;
```

```
In file included from kvrocks/src/search/hnsw_indexer.cc:21:
kvrocks/src/search/hnsw_indexer.h:81:1: warning: 'HnswIndex' defined as a struct here but previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
struct HnswIndex {
^
/home/skyjiang/c-project/kvrocks/src/search/hnsw_indexer.h:34:1: note: did you mean struct here?
class HnswIndex;
^~~~~
```

### Solution

Do we need to resolve these warning? Make the compilation log clean.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/src/commands/cmd_cluster.cc
+++ b/src/commands/cmd_cluster.cc
@@ -311,7 +311,6 @@ class CommandClusterX : public Commander {
   std::string nodes_str_;
   std::string dst_node_id_;
   int64_t set_version_ = 0;
-  int64_t slot_ = -1;
   std::vector<SlotRange> slot_ranges_;
   bool force_ = false;
 

--- a/src/search/hnsw_indexer.h
+++ b/src/search/hnsw_indexer.h
@@ -31,7 +31,7 @@
 
 namespace redis {
 
-class HnswIndex;
+struct HnswIndex;
 
 struct HnswNode {
   using NodeKey = std::string;
@@ -51,7 +51,7 @@ struct HnswNode {
                       rocksdb::WriteBatchBase* batch) const;
   Status RemoveNeighbour(const NodeKey& neighbour_key, const SearchKey& search_key, engine::Storage* storage,
                          rocksdb::WriteBatchBase* batch) const;
-  friend class HnswIndex;
+  friend struct HnswIndex;
 };
 
 struct VectorItem {

